integrations/sonarqube/client.py

```diff
@@ -132,22 +149,47 @@ async def send_paginated_api_request(
             logger.error(f"HTTP occurred while fetching paginated data: {e}")
             raise
 
+    def process_selector_query_params(self, selector: Selector) -> dict[str, Any]:
+        if hasattr(selector, "api_query_params") and selector.api_query_params:
+            return selector.api_query_params.generate_request_params()
+        return {}
```

Comment:
Can u instead just override the resourceConfig implementation and inherit this functionality to exists in every of the kinds?

```
class MY_BASE_CUSTOM_SELECTOR(Selector):
    def generate_request_params(self):
        return {}
        
class MY_CUSTOM_SELECTOR(MY_BASE_CUSTOM_SELECTOR):
    ...

class MY_SONAR_CUSTOM_RESOURCE(ResourceConfig):
    selector: MY_CUSTOM_SELECTOR
        
class B(MY_SONAR_CUSTOM_RESOURCE):
  ...